axon-framework/events/event-processors/README.md

```diff
@@ -253,12 +308,213 @@ public interface ErrorHandler {
+
+```java
+public class DeadletterProcessor {
+
+    private final EventProcessingConfiguration eventProcessingConfiguration;
+
+    public DeadletterProcessor(EventProcessingConfiguration eventProcessingConfiguration) {
+        this.eventProcessingConfiguration = eventProcessingConfiguration;
+    }
+
+    public void repairErrorEvent() {
+        eventProcessingConfiguration.sequencedDeadLetterProcessor(PROCESSING_GROUP)
+                                    .ifPresent(p -> p.process(deadLetter -> deadLetter.message()
+                                                                                      .getPayload() instanceof ErrorEvent));
+    }
+}
+```
+
+{% endtab %}
+
+{% tab title="Process any dead-letter in the queue" %}
+
+```java
+public class DeadletterProcessor {
+
+    private final EventProcessingConfiguration eventProcessingConfiguration;
+
+    public DeadletterProcessor(EventProcessingConfiguration eventProcessingConfiguration) {
+        this.eventProcessingConfiguration = eventProcessingConfiguration;
+    }
+
+    public void repairAnyEvent() {
+        eventProcessingConfiguration.sequencedDeadLetterProcessor(PROCESSING_GROUP)
+                                    .ifPresent(SequencedDeadLetterProcessor::processAny);
+    }
+}
+```
+
+{% endtab %}
+
+{% tab title="Process all dead-letters in the queue" %}
+
+```java
+public class DeadletterProcessor {
+
+    private final EventProcessingConfiguration eventProcessingConfiguration;
+
+    public DeadletterProcessor(EventProcessingConfiguration eventProcessingConfiguration) {
+        this.eventProcessingConfiguration = eventProcessingConfiguration;
+    }
+
+    public void repairAllEvents() {
+        Optional<Iterable<Iterable<DeadLetter<? extends EventMessage<?>>>>> deadLetters = eventProcessingConfiguration.deadLetterQueue(
+                PROCESSING_GROUP).map(SequencedDeadLetterQueue::deadLetters);
+
+        deadLetters.ifPresent(sequences -> {
+            // Iterate over all sequences
+            for (Iterable<DeadLetter<? extends EventMessage<?>>> sequence : sequences) {
+                // Iterate over all dead-letters belonging to the same sequence
+                for (DeadLetter<? extends EventMessage<?>> deadLetterInSequence : sequence) {
+                    eventProcessingConfiguration.sequencedDeadLetterProcessor(PROCESSING_GROUP).ifPresent(
+                            eventMessageSequencedDeadLetterProcessor -> {
+                                eventMessageSequencedDeadLetterProcessor.process(
+                                        deadLetter -> deadLetter.message().getIdentifier()
+                                                                .equals(deadLetterInSequence.message()
+                                                                                            .getIdentifier()));
+                            });
+                }
+            }
+        });
+    }
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
```

Comment:
Enter section:
```suggestion
#### Dead-Letter Enqueue Policy
```
I am least certain about this section header.
Perhaps "Deciding when to Enqueue" is clearer to the reader.